### ISSUE TITLE
Look into supporting pymunk.Space for collision

### ISSUE THREAD

**Issue by einarf** [pythonarcade/arcade#1297]
Pymunk has a pretty neat collision system using spaces and groups. It supports handlers/callbacks to notify about collision. I strongly suspect this can scale much better the generic sprite collision we're currently using in many cases. Especially of you're doing N x M collision checks and can't use spatial hashing due to both spritelists being highly dynamic.

**Comment by eschan145** [comment]
Yeah, it's much faster than the arcade one. It;s object-oriented, so you can get additional properties and features. Using this with my medieval wargame, I got much better performance than using the arcade collision checks. We could also make object-oriented collision checks for arcade too.

**Comment by DragonMoffon** [comment]
I'd just like to add to this that the performance improvement we get from pymunk also games down to the fact that it isn't pure python. Def think we should expose it, but I'm also  for making pymunk an optional import.

When I do my by physics engine overhaul post 3.0 this will be part of my todo.
